pkg/peertls/extensions/revocations.go

```diff
@@ -116,14 +103,13 @@ func revocationChecker(opts *Options) HandlerFunc {
 			return nil
 		}
 
-		caHash := pkcrypto.SHA256Hash(ca.Raw)
 		leafHash := pkcrypto.SHA256Hash(leaf.Raw)
 
 		// NB: we trust that anything that made it into the revocation DB is valid
 		//		(i.e. no need for further verification)
 		switch {
-		case bytes.Equal(lastRev.CertHash, caHash):
-			return ErrRevokedCert
+		case lastRev.Final:
```

Comment:
why does the fallthrough happen here?

Reply:
If the revocation is final then we consider any certificate chain with the same node ID as revoked. I'm not sure if this is necessarily the right way to do this. I think that we want all CA revocations to always be final and this implementation would allow for a user to create a non-final CA revocation. We should probably be keeping a hash of the public key of the certificate that's being revoked instead of a hash of the certificate (what we're currently doing).